src/trace/trace.c

```diff
@@ -499,6 +499,11 @@ void trace_off(void)
 void trace_init(struct sof *sof)
 {
 	sof->trace = rzalloc(SOF_MEM_FLAG_USER | SOF_MEM_FLAG_COHERENT, sizeof(*sof->trace));
+	if (!sof->trace) {
+		mtrace_printf(LOG_LEVEL_ERROR, "trace_init(): allocation failed");
+		sof_panic(SOF_IPC_PANIC_IPC);
```

Comment:
do we need to panic here?

Reply:
This is quite a bad situation, as there be no fats way to know where the failure as there may be no trace data so best to give up early.